server/graphql/common/expenses.ts

```diff
@@ -382,11 +382,36 @@ export const canApprove: ExpensePermissionEvaluator = async (req, expense, optio
     return false;
   } else {
     expense.collective = expense.collective || (await req.loaders.Collective.byId.load(expense.CollectiveId));
-    if (hasPolicy(expense.collective, POLICIES.EXPENSE_AUTHOR_CANNOT_APPROVE) && req.remoteUser.id === expense.UserId) {
+    expense.collective.host =
+      expense.collective.host || (await req.loaders.Collective.byId.load(expense.collective.HostCollectiveId));
```

Comment:
@hdiniz it is valid for a collective to receive expenses without being connected to a fiscal host yet, we should ignore if the `HostCollectiveId` is null, or if `approvedAt` is null. I believe this is causing the following error: https://sentry.io/organizations/open-collective/issues/3814083338/?project=5199682